pkg/cedar/core/client.go

```diff
@@ -154,3 +158,123 @@ func (c *Client) GetSystem(ctx context.Context, id string) (*models.CedarSystem,
 		SystemMaintainerOrgComp: responseArray[0].SystemMaintainerOrgComp,
 	}, nil
 }
+
+// GetDeploymentsOptionalParams represents the optional parameters that can be used to filter deployments
+// when searching through the CEDAR API
+// TODO - is this how we want to handle multiple optional parameters?
+type GetDeploymentsOptionalParams struct {
+	DeploymentType null.String
+	State          null.String
+	Status         null.String
+}
+
+// GetDeployments makes a GET call to the /deployment endpoint
+func (c *Client) GetDeployments(ctx context.Context, systemID string, optionalParams *GetDeploymentsOptionalParams) ([]*models.CedarDeployment, error) {
+	if !c.cedarCoreEnabled(ctx) {
+		appcontext.ZLogger(ctx).Info("CEDAR Core is disabled")
+		return []*models.CedarDeployment{}, nil
+	}
+
+	// Construct the parameters
+	params := apideployments.NewDeploymentFindListParams()
+	params.SetSystemID(systemID)
+	params.HTTPClient = c.hc
+
+	if optionalParams != nil {
+		if optionalParams.DeploymentType.Ptr() != nil {
+			params.SetDeploymentType(optionalParams.DeploymentType.Ptr())
+		}
+
+		if optionalParams.State.Ptr() != nil {
+			params.SetState(optionalParams.State.Ptr())
+		}
+
+		if optionalParams.Status.Ptr() != nil {
+			params.SetStatus(optionalParams.Status.Ptr())
+		}
+	}
+
+	// Make the API call
+	resp, err := c.sdk.Deployment.DeploymentFindList(params, c.auth)
+	if err != nil {
+		return []*models.CedarDeployment{}, err
+	}
+
+	if resp.Payload == nil {
+		return []*models.CedarDeployment{}, fmt.Errorf("no body received")
+	}
+
+	responseArray := resp.Payload.Deployments
+
+	if len(responseArray) == 0 {
+		return nil, &apperrors.ResourceNotFoundError{Err: fmt.Errorf("no deployments found"), Resource: []*models.CedarDeployment{}}
+	}
+
+	// Convert the auto-generated struct to our own pkg/models struct
+	retVal := []*models.CedarDeployment{}
+
+	// Populate the Deployment field by converting each item in resp.Payload.Deployments
+	// generated swagger client turns JSON nulls into Go zero values, so use null/zero package to convert them back to nullable values
```

Comment:
This is due to an issue with the Swagger spec for CEDAR. CEDAR uses Swagger 2.0, which doesn't have the `nullable` attribute that was added in OpenAPI 3.0; furthermore, the go-swagger package we're using for generating the client only works with Swagger 2.0. As the comment says, the go-swagger generated code turns `null` values in the JSON that CEDAR returns into zero values (empty strings, generally), so I used the [`null/zero`](https://pkg.go.dev/github.com/guregu/null/zero) package to nullable values in Go. The generated GraphQL code then turns Go `nil` values into GraphQL `null` values, which seems like the appropriate thing to do.

That said, I'm not _positive_ that this is the best approach, so I'd appreciate feedback.

Reply:
I think this makes sense overall for the String cases. I'd be hesitant to try and parse any other zero values (like integers), but I can't imagine treating "" as `nil -> null` should be an issue.

On the other hand (I always try and look at both angles) it also probably doesn't really do any harm to return "", especially because this is going to all be parsed and used by a JS frontend, which treats "" as falsy, and probably would render `null` and `""` the same.

Let's keep it around, since `zero.StringFrom` and `zero.TimeFrom` seem like their whole purpose is to never fail, so worst case is we come back and touch this up to just return empty strings (I can only really imagine this happening on the `zero.TimeFrom` calls)